lib/yt_search/subtitles.ex

```diff
@@ -28,26 +27,56 @@ defmodule YtSearch.Subtitle do
     SubtitleRepo.replica().all(query)
   end
 
+  def blob(nil), do: nil
+
+  def blob(%__MODULE__{} = subtitle) do
+    blob(subtitle_id(subtitle))
+  end
+
+  def blob(id) when is_bitstring(id) do
+    case File.read(path_for(id)) do
+      {:ok, data} -> data
+      {:error, :enoent} -> nil
+    end
+  end
+
+  def path_for(id) do
```

Comment:
`path_for` can be a function that receives `%__MODULE__{}` so you don't need to do `path_for(subtitle_id(sub))` (plus ensures we never make paths out of any strings, including invalid strings)